src/Npgsql/PostgresException.cs

```diff
@@ -293,22 +290,44 @@ public override IDictionary Data
         {
             get
             {
-                // Remarks: return Dictionary with object keys although all our keys are string keys
-                // because System.Windows.Threading.Dispatcher relies on that
-                return _data ?? (_data = (
-                    from p in typeof(PostgresException).GetProperties()
-                    let k = p.Name
-                    where p.Name != nameof(Data)
-                    where p.GetCustomAttribute<PublicAPIAttribute>() != null
-                    let v = p.GetValue(this)
-                    where v != null
-                    where k != nameof(Position) && k != nameof(InternalPosition) || (int)v != 0
-                    select new { Key = k, Value = v }
-                    ).ToDictionary(kv => (object)kv.Key, kv => kv.Value)
-                );
+                if (_data == null)
```

Comment:
nit: invert to reduce nesting

Reply:
And have two returns?

Reply:
I tend to prefer an early exit pattern over unitary returns, but that's a personal preference. If you don't think it improves the readability, feel free to disregard.

Reply:
It makes sense in some cases. For example, when you have a lot of conditional returns. In such situations I prefer that pattern too, but not here (: